test/test_e2e.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+from itertools import chain
+import json
+from pathlib import Path
+import subprocess
+import pytest
+
+ABANDONING_PARENT = str(Path(__file__).with_name("data") / "abandoning_parent.sh")
+
+
+def test_sanity(temp_output_dir: str) -> None:
+    command = f"duct -p {temp_output_dir}log_ sleep 0.1"
+    subprocess.check_output(command, shell=True)
+
+
+@pytest.mark.parametrize("num_children", [1, 10, 25, 101])
+def test_abandoning_parent(temp_output_dir: str, num_children: int) -> None:
+    duct_prefix = f"{temp_output_dir}log_"
+    command = f"duct --s-i 0.001 --r-i 0.01 -p {duct_prefix} {ABANDONING_PARENT} {num_children} sleep 0.35"
+    subprocess.check_output(command, shell=True)
+
+    with open(f"{duct_prefix}usage.json") as usage_file:
+        all_samples = [json.loads(line) for line in usage_file]
+
+    all_pids = {
+        pid
+        for pid in chain.from_iterable(sample["processes"] for sample in all_samples)
+    }
```

Comment:
```suggestion
    all_pids = set(chain.from_iterable(sample["processes"] for sample in all_samples))
```
?

Reply:
someone really obnoxious could have even made it smth like ( I didn't test, but should work)

```suggestion
    all_pids = set(chain.from_iterable(map(operator.attrgetter("processes"), all_samples)))
```
but that is pushing it close/over the sensible limit in aim for de-duplication ;-)

Reply:
I went with option 1 for readability, but the attrgetter with map is cool :)